tools/cli/commands/clean.js

```diff
@@ -0,0 +1,607 @@
+/**
+ * External dependencies
+ */
+import chalk from 'chalk';
+import inquirer from 'inquirer';
+import child_process from 'child_process';
+
+/**
+ * Internal dependencies
+ */
+import promptForProject, { promptForType } from '../helpers/promptForProject';
+import { normalizeCleanArgv } from '../helpers/normalizeArgv';
+import { runCommand } from '../helpers/runCommand';
+import { allProjects } from '../helpers/projectHelpers';
+import fs from 'fs';
+
+/**
+ * Command definition for the build subcommand.
+ *
+ * @param {object} yargs - The Yargs dependency.
+ *
+ * @returns {object} Yargs with the build commands defined.
+ */
+export function cleanDefine( yargs ) {
+	yargs.command(
+		'clean [project]',
+		'Removes unversioned files and folder from a specific project.',
+		yarg => {
+			yarg
+				.positional( 'project', {
+					describe: 'Project in the form of type/name, e.g. plugins/jetpack',
+					type: 'string',
+				} )
+				.positional( 'include', {
+					alias: 'i',
+					describe: 'Files/folders to include for deletion',
+					type: 'string',
+					choices: [ 'node_modules', 'composer.lock', 'vendor', 'working', 'ignored' ],
+				} )
+				.option( 'all', {
+					alias: 'a',
+					type: 'boolean',
+					description: 'Remove everything from monorepo root',
+				} )
+				.option( 'dist', {
+					type: 'boolean',
+					description: 'Remove distributed files (vendor, node_modules)',
+				} );
+		},
+		async argv => {
+			await cleanCli( argv );
+			if ( argv.v ) {
+				console.log( argv );
+			}
+		}
+	);
+
+	return yargs;
+}
+
+/**
+ * Handle args for clean command.
+ *
+ * @param {argv}  argv - the arguments passed.
+ */
+export async function cleanCli( argv ) {
+	if ( argv.project ) {
+		await parseProj( argv );
+	} else {
+		await promptProj( argv );
+	}
+
+	if ( argv.include ) {
+		await parseToClean( argv );
+	} else {
+		await promptForClean( argv );
+	}
+
+	const allFiles = await collectAllFiles( argv.toClean );
+	const toCleanFiles = await collectCleanFiles( allFiles, argv.toClean );
+	const runConfirm = await confirmRemove( argv );
+
+	if ( ! runConfirm.confirm ) {
+		console.log( chalk.red( 'Cancelling jetpack clean.' ) );
+		return;
+	}
+
+	await cleanFiles( toCleanFiles, argv );
+	return;
+}
+
+/**
+ * Delete the files that we want.
+ *
+ * @param {Array} toCleanFiles - files that we want to clean.
+ * @param {object} argv - the arguments passed.
+ */
+async function cleanFiles( toCleanFiles, argv ) {
+	for ( const file of toCleanFiles ) {
+		fs.rm( file, { recursive: true, force: true }, ( err ) => {
+			console.log( 'Cleaning ', file );
+			if ( err ) {
+				// File deletion failed
+				console.error( err.message );
+				return;
+			}
+		} );
```

Comment:
Now that I look at this closer, the asynchronity here may be problematic.
```suggestion
		console.log( `Cleaning ${ file }` );
		try {
			fs.rmSync( file, { recursive: true, force: true } );
		} catch ( e ) {
			// Although it looks like rmSync doesn't do any error reporting, it probably should.
			console.error( chalk.red( err.message ) );
			process.exit( 1 );
		}
```

Reply:
Nice catch with this! Should be resolved in b7bc3d7